src/libraries/Common/src/Interop/Unix/System.Native/Interop.GNameUName.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Buffers;
+using System.Text;
+using System;
+using System.Collections.Generic;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Gets the group name associated to the specified group ID.
+        /// </summary>
+        /// <param name="gid">The group ID.</param>
+        /// <returns>On success, return a string with the group name. On failure, returns a null string.</returns>
+        internal static string GetGName(uint gid) => GetGNameOrUName(gid, isGName: true);
+
+        /// <summary>
+        /// Gets the user name associated to the specified user ID.
+        /// </summary>
+        /// <param name="uid">The user ID.</param>
+        /// <returns>On success, return a string with the user name. On failure, returns a null string.</returns>
+        internal static string GetUName(uint uid) => GetGNameOrUName(uid, isGName: false);
+
+        private static string GetGNameOrUName(uint id, bool isGName)
+        {
+            int bufferSize = 256; // Upper limit allowed for login name in kernel
+
+            Span<byte> buffer = stackalloc byte[bufferSize];
+
+            int resultLength = GetGNameOrUnameInternal(id, isGName, buffer);
+            if(resultLength > 0 && resultLength <= buffer.Length)
+            {
+                return Encoding.UTF8.GetString(buffer.Slice(0, resultLength));
+            }
+
+            while (true)
+            {
+                bufferSize *= 2;
+                byte[] pooledBuffer = ArrayPool<byte>.Shared.Rent(bufferSize);
+                try
+                {
+                    resultLength = GetGNameOrUnameInternal(id, isGName, pooledBuffer);
+                    if(resultLength > 0 && resultLength <= pooledBuffer.Length)
+                    {
+                        return Encoding.UTF8.GetString(pooledBuffer.AsSpan(0, resultLength));
+                    }
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(pooledBuffer);
+                }
+
+                if (bufferSize >= 1024)
+                {
+                    return string.Empty;
+                }
+            }
```

Comment:
if 1024 bytes is the maximum, we can imo just stack-allocate it and don't use pool at all

```suggestion
            Span<byte> buffer = stackalloc byte[1024];

            int resultLength = GetGNameOrUnameInternal(id, isGName, buffer);
            if(resultLength > 0 && resultLength <= buffer.Length)
            {
                return Encoding.UTF8.GetString(buffer.Slice(0, resultLength));
            }
            return string.Empty;
```

Reply:
@GrabYourPitchforks @jkotas do we have any policy on how many bytes we can allocate on the stack? I remember seeing 1024 in a few places.

Reply:
There is no hard and fast rule for this. `1024` is on the high side, but it is still ok - especially if you know that you do not need more.

Reply:
> if 1024 bytes is the maximum, we can imo just stack-allocate it and don't use pool at all

It's not the limit (https://github.com/dotnet/runtime/pull/69107/files#r871098720 and https://github.com/dotnet/runtime/pull/69107/files#r871097622).